src/modules/adaptors/Convex/ConvexCurveAdaptor.sol

```diff
@@ -0,0 +1,333 @@
+     * @return whether adaptor returns debt or not
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to deposit and stake LPTs into Convex markets via the respective Convex market Booster contract
+     * @param _pid specified pool ID corresponding to LPT convex market
+     * @param _amount amount of LPT to deposit and stake
+     * NOTE: stake bool in `boosted.deposit()` function to stake assets is set to always true for strategist calls
+     */
+    function depositLPTInConvexAndStake(
+        uint256 _pid,
+        address _baseRewardPool,
+        ERC20 _lpt,
+        CurvePool _pool,
+        bytes4 _selector,
+        uint256 _amount
+    ) public {
+        _validatePositionIsUsed(_pid, _baseRewardPool, _lpt, _pool, _selector); // validate pid representing convex market within respective booster
+        _amount = _maxAvailable(_lpt, _amount);
+
+        _lpt.approve(address(booster), _amount);
+        booster.deposit(_pid, _amount, true);
+        _revokeExternalApproval(_lpt, address(booster));
+    }
+
+    /**
+     * @notice Allows strategists to withdraw from Convex markets via Booster contract w/ or w/o claiming rewards
+     * NOTE: this adaptor will always unwrap to CRV LPTs if possible. It will not keep the position in convex wrapped LPT position.
+     * NOTE: If _claim is true, this claims all rewards associated to a cellar interacting w/ Convex markets. The BaseRewardPool contract has the function for withdrawing and unwrapping whilst also claiming all rewards. NOTE: if it does not claim all extra rewards (say another reward contract is linked to it somehow), then we can just make other adaptors that handle that.
+     * @param _baseRewardPool for respective convex market (w/ trusted poolId)
+     * @param _amount of LPTs to unstake, unwrap, and withdraw from convex market to calling cellar
+     * @param _claim whether or not to claim all rewards from BaseRewardPool
+     */
+    function withdrawFromBaseRewardPoolAsLPT(address _baseRewardPool, uint256 _amount, bool _claim) public {
+        IBaseRewardPool baseRewardPool = IBaseRewardPool(_baseRewardPool);
+
+        if (_amount == type(uint256).max) {
+            _amount = baseRewardPool.balanceOf(address(this));
+        }
+        baseRewardPool.withdrawAndUnwrap(_amount, _claim);
+    }
+
+    /**
+     * @notice Allows strategists to get rewards for an Convex Booster without withdrawing/unwrapping from Convex market
+     * @param _pid specified pool ID corresponding to LPT convex market
+     * @param _baseRewardPool for respective convex market (w/ trusted poolId)
+     * @param _claimExtras Whether or not to claim extra rewards associated to the Convex booster (outside of rewardToken for Convex booster)
+     */
+    function getRewards(
```

Comment:
I feel like we dont need to validate if the position is used, since we are not giving any approvals.

Reply:
Sounds good, I've removed it.